indexing-service/src/main/java/io/druid/indexing/common/task/IndexTaskUtils.java

```diff
@@ -36,6 +36,8 @@
 
 public class IndexTaskUtils
 {
+  public static final int DEFAULT_MAX_ROWS_IN_MEMORY = 75_000;
```

Comment:
Let's use `1_000_000` here.

Reply:
Now that I think about it some more, this class isn't going to work for these constants.

I'm hoping we can just write them in one place and have other stuff reference them. They are both magical and it's good for magic to be rare. But, one of the things (for example) that has to reference them is HadoopTuningConfig, which is in the hadoop-indexer package.

Maybe the right place to put these is in `TuningConfig` itself (i.e. io.druid.segment.indexing.TuningConfig).

Reply:
ok, yeah, when i added the defaults to `IndexTaskUtils` it didn't cover all the places I had defined this default. Defining those in `TuningConfig` seems correct. But I am also checking for `maxBytesInMemory == 0` in `OnheapIncrementalIndex` and setting it to default if 0, and now that seems wrong. May be I should instead throw an exception in `IncrementalIndex.buildOnHeap`. Perhaps now I understand what @jihoonson was talking about.